launch/launch/launcher.py

```diff
@@ -225,7 +254,18 @@ def _run(self):
                         except ProcessLookupError:
                             pass
 
-            yield from asyncio.wait(all_futures.keys())
+            kwargs = {}
+            if not isinstance(threading.current_thread(), threading._MainThread):
+                # wake up periodically if we are not in the main thread
```

Comment:
I think that's the whole point of this check, if you're not calling `launch()` in the main thread then it needs to do something special. I don't know what you mean by "I would expect the check for _MainThread to not work in this case.", do you mean it will give a false positive or a false negative?